src/Development/IDE/Core/Compile.hs

```diff
@@ -690,11 +626,13 @@ parseFileContents env customPreprocessor dflags comp_pkgs filename modTime conte
 
                -- Ok, we got here. It's safe to continue.
                let IdePreprocessedSource preproc_warns errs parsed = customPreprocessor rdr_module
-               unless (null errs) $ throwE $ diagFromStrings "parser" DsError errs
-               let parsed' = removePackageImports comp_pkgs parsed
+
+               unless (null errs) $
+                  throwE $ diagFromStrings "parser" DsError errs
+
                let preproc_warnings = diagFromStrings "parser" DsWarning preproc_warns
-               ms <- getModSummaryFromBuffer filename modTime dflags parsed' contents
```

Comment:
@mpickering need your help here. Does `removePackageImports` need to be called before we compute the `ModSummary`? Because we don't recompute the `ModSummary` here anymore, we just re-use the one we computed before.

Reply:
My take on this: since we don't allow GHC to resolve imports anymore, the only place where import resolution happens is in our code, namely in `GetLocatedImportsRule`. So the removal of package imports can be inlined there in order to preserve the AST and original imports as much as possible. Otherwise plugins like the HLS import lens or code actions which use the AST might end up "losing" package imports accidentally.

Reply:
OK, but I'm confused. `GetLocatedImportsRule` uses `GetModSummaryWithoutTimestamps`, which doesn't call `removePackageImports`, so it should never have been affected by it. The only things which are affected by `removePackageImports` are things which use the modsummary from the parsed module.

Reply:
Back in the day when `removePackageImports` was added, `GetModSummary` didn't exist and `GetLocatedImportsRule` still used the parsed AST to extract the imports.

Reply:
Sounds like we need a testcase. How do I invoke this code path? Will a `multi-cradle` with a file using `PackageImports` to import a conflicting module from two different packages do it?

Reply:
Looks like we can just remove it, since the logic is already present in findImports: https://github.com/haskell/ghcide/blob/master/src/Development/IDE/Import/FindImports.hs#L126

Reply:
Test case is the `generic-lens` repo. I think the situation was as you described (conflicting module from two different packages)